### ISSUE TITLE
Button is broken in webplayer

### ISSUE THREAD

**Issue by Lance3d** [oddgames/UIToolkit#116]
Button does not respond in webplayer. Any demo scene can reproduce this problem.

**Comment by Lance3d** [comment]
PS, I am using Unity 3.5.3

**Comment by Muxeu** [comment]
I also have this bug, OSX, Andriod, Windows and other works fine, but webplayer not.

**Comment by Muxeu** [comment]
The bug appear when UITouchMaker::createTouchFromInput trying to modify private properties of Touch. This works on all platforms, but webplayer throw error.

I create own UITouch wich can contains Touch or fake touch from input. Now webplayer works fine.

Sorry for my bad english

**Comment by varomix** [comment]
I'm having the same issue, can you provide and example of what you did to solve it please
thank you

**Comment by rotorz** [comment]
Same issue here, any ideas?

**Comment by varomix** [comment]
we didn't had time to hunt bugs so we end up adding transparent standard button on top of our buttons for the web version, super dirty but works if you don't have tons of buttons :(

**Comment by rotorz** [comment]
Thanks for the advice, I will just do that for now.

**Comment by NigeDemo** [comment]
Also struggling with this issue - unfortunately varomixs' quick'n'dirty fix won't work for me. Has anyone got any pointers / concrete advice on how you could get around this issue??

Thanks in advance

**Comment by hyakugei** [comment]
FWIW - i've got a fix, using the same, or very similar, technique that @Muxeu used. This fix is on my branch of the UIToolkit. Feel free to grab it. Please note, i've not done extensive (any!) testing on all the different platforms, or examined how this fix impacts CPU time. For me, it has been working fine in the web & iOS players.

**Comment by varomix** [comment]
oh that is awesome Jos

thanks

**Comment by NigeDemo** [comment]
That is fantastic Jo, thanks for sharing that.

**Comment by rotorz** [comment]
Thanks Jos, works like a charm!

**Comment by phfatmonkey** [comment]
I had to use GUI Texture buttons set underneath UIToolkit buttons to get this to work on Web Player. Works fine, and it's an easy enough fix, However, I've found that the UIToolKit UI doesn't display on browsers on OSX. Anyone else have this problem?

**Comment by NigeDemo** [comment]
Yes I've found the OSX browser thing too - odd. Any ideas?? Can't stand working with UnityGUI just for web builds :(
